adafruit_si1145.py

```diff
@@ -132,47 +139,47 @@ def uv_index_enabled(self, enable):
         self._uv_index_enabled = enable
 
     @property
-    def uv_index(self):
+    def uv_index(self) -> float:
         """The UV Index value"""
         self._send_command(_CMD_ALS_FORCE)
         return self._aux_data[0] / 100
 
-    def reset(self):
+    def reset(self) -> None:
         """Perform a software reset of the firmware."""
         self._send_command(_CMD_RESET)
         time.sleep(0.05)  # doubling 25ms datasheet spec
 
-    def clear_error(self):
+    def clear_error(self) -> None:
         """Clear any existing error code."""
         self._send_command(_CMD_NOP)
 
-    def _param_query(self, param):
+    def _param_query(self, param: int) -> int:
         self._send_command(_CMD_PARAM_QUERY | (param & 0x1F))
         return self._read_register(_PARAM_RD)
 
-    def _param_set(self, param, value):
+    def _param_set(self, param: int, value: int) -> None:
         self._write_register(_PARAM_WR, value)
         self._send_command(_CMD_PARAM_SET | (param & 0x1F))
 
-    def _send_command(self, command):
+    def _send_command(self, command: int) -> int:
         counter = self._read_register(_RESPONSE) & 0x0F
         self._write_register(_COMMAND, command)
         if command in (_CMD_NOP, _CMD_RESET):
             return 0
         response = self._read_register(_RESPONSE)
         while counter == response & 0x0F:
             if response & 0xF0:
-                raise RuntimeError("SI1145 Error: 0x{:02x}".format(response))
+                raise RuntimeError(f"SI1145 Error: {response:#x}")
             response = self._read_register(_RESPONSE)
         return response
 
-    def _read_register(self, register, length=1):
+    def _read_register(self, register: int, length: int = 1) -> Union[int, bytearray]:
         buffer = bytearray(length)
         with self.i2c_device as i2c:
             i2c.write_then_readinto(bytes([register]), buffer)
         return buffer[0] if length == 1 else buffer
 
-    def _write_register(self, register, buffer):
+    def _write_register(self, register: int, buffer: ReadableBuffer) -> None:
```

Comment:
It looks like `buffer` can also be an `int`.  I also don't think that `ReadableBuffer` is okay here because `memoryview` won't work with the addition, I believe.  Let's type it as `Union[int, bytes, bytearray]` instead.